packages/cli/src/config/utils.ts

```diff
@@ -11,3 +11,23 @@ export const ensureStringArray = (values: string[], { env }: SchemaObj<string>)
 		if (typeof value !== 'string') throw new NotStringArrayError(env);
 	}
 };
+
+export function findSchemaEnvVars(schema: Record<string, unknown>) {
+	const envVars: string[] = [];
+
+	const traverse = (obj: Record<string, unknown>) => {
+		for (const key in obj) {
+			const value = obj[key] as Record<string, unknown> | string;
+
+			if (typeof value === 'object') {
+				traverse(value);
+			} else if (typeof value === 'string' && key === 'env') {
+				envVars.push(value);
+			}
+		}
+	};
+
+	traverse(schema);
```

Comment:
Why do we create a function and call it right away? Would something break if we lifted the body of traverse into findSchemaEnvVars' body?